tensorboard/plugins/audio/summary.py

```diff
@@ -0,0 +1,202 @@
+    raise ValueError('Unknown file_format: %r' % file_format)
+
+  with tf.name_scope(name), \
+       tf.control_dependencies([tf.assert_rank(audio, 3)]):
+    limited_audio = audio[:max_outputs]
+    encoded_audio = tf.map_fn(encoder, limited_audio,
+                              dtype=tf.string,
+                              name='encode_each_audio')
+    if labels is None:
+      limited_labels = tf.tile([''], tf.shape(limited_audio)[:1])
+    else:
+      limited_labels = labels[:max_outputs]
+    tensor = tf.transpose(tf.stack([encoded_audio, limited_labels]))
+    summary_metadata = metadata.create_summary_metadata(
+        display_name=display_name,
+        description=description,
+        encoding=encoding)
+    return tf.summary.tensor_summary(name='audio_summary',
+                                     tensor=tensor,
+                                     collections=collections,
+                                     summary_metadata=summary_metadata)
+
+
+def pb(name,
+       audio,
+       sample_rate,
+       labels=None,
+       max_outputs=3,
+       file_format=None,
+       display_name=None,
+       description=None):
+  """Create an audio summary protobuf.
+
+  This behaves as if you were to create an `op` with the same arguments
+  (wrapped with constant tensors where appropriate) and then execute
+  that summary op in a TensorFlow session.
+
+  Arguments:
+    name: A unique name for the generated summary node.
+    audio: An `np.array` representing audio data with shape `[k, t, c]`,
+      where `k` is the number of audio clips, `t` is the number of
+      frames, and `c` is the number of channels. Elements should be
+      floating-point values in `[-1.0, 1.0]`.
+    sample_rate: An `int` that represents the sample rate, in Hz.
+      Must be positive.
+    labels: Optional list (or rank-1 `np.array`) of textstrings or UTF-8
+      bytestrings whose length is the first dimension of `audio`, where
+      `labels[i]` contains arbitrary textual information about
+      `audio[i]`. (For instance, this could be some text that a TTS
+      system was supposed to produce.) Markdown is supported.
+    max_outputs: Optional `int`. At most this many audio clips will be
+      emitted. When more than `max_outputs` many clips are provided, the
+      first `max_outputs` many clips will be used and the rest silently
+      discarded.
+    file_format: A constant `str` indicating the desired encoding. You
+      can choose any format you like, as long as it's "wav". Please see
+      the "API compatibility note" below.
+    display_name: Optional name for this summary in TensorBoard, as a
+      `str`. Defaults to `name`.
+    description: Optional long-form description for this summary, as a
+      `str`. Markdown is supported. Defaults to empty.
+
+  Returns:
+    A `tf.Summary` protobuf object.
+
+  API compatibility note: The default value of the `file_format`
+  argument is _not_ guaranteed to remain unchanged across TensorBoard
```

Comment:
I recommend stating that it *will* change to FLAC in the future, in no uncertain terms.